src/shogun/distance/EuclideanDistance.cpp

```diff
@@ -177,3 +179,61 @@ float64_t CEuclideanDistance::distance_upper_bounded(int32_t idx_a, int32_t idx_
 
 	return result;
 }
+
+SGMatrix<float64_t> CEuclideanDistance::get_distance_matrix()
+{
+	if (lhs->get_feature_class()==C_DENSE && rhs->get_feature_class()==C_DENSE)
+	{
+		if (lhs->get_feature_type()==F_DREAL && rhs->get_feature_type()==F_DREAL)
+		{
+			CDenseFeatures<float64_t>* casted_lhs=static_cast<CDenseFeatures<float64_t>*>(lhs);
+			CDenseFeatures<float64_t>* casted_rhs=static_cast<CDenseFeatures<float64_t>*>(rhs);
+			return compute_distance_matrix(casted_lhs, casted_rhs);
+		}
+		else if (lhs->get_feature_type()==F_SHORTREAL && rhs->get_feature_type()==F_SHORTREAL)
+		{
+
+			CDenseFeatures<float32_t>* casted_lhs=static_cast<CDenseFeatures<float32_t>*>(lhs);
+			CDenseFeatures<float32_t>* casted_rhs=static_cast<CDenseFeatures<float32_t>*>(rhs);
+			return compute_distance_matrix(casted_lhs, casted_rhs);
+		}
+		else
+			return CDistance::get_distance_matrix();
+	}
+	else
+		return CDistance::get_distance_matrix();
+}
+
+template <typename T>
+SGMatrix<float64_t> CEuclideanDistance::compute_distance_matrix(CDenseFeatures<T>* casted_lhs, CDenseFeatures<T>* casted_rhs)
+{
+	int32_t n = casted_lhs->get_num_vectors();
+	int32_t m = casted_rhs->get_num_vectors();
+
+	MatrixXd map_result;
```

Comment:
Hey @Saurabh7 sorry for being annoying again but just a minor comment. Isn't it possible to use the `SGMatrix<float64_t> result` matrix from here itself and then use an Eigen map for it for operations that follow? If operator `=` does a deep copy for Eigen (required in the expression `map_result = (map_l.transpose()*map_r).template cast<double>();`, then we're good, right? Saves an extra alloc and copy. Can you please check once whether it's doable?